infra/lambda/datasets-endpoint/test_function.py

```diff
@@ -0,0 +1,307 @@
+"""
+Basic Lambda function tests. Working Data Lake AWS environment is required
+(run '$ cdk deploy' before running tests).
+"""
+
+import logging
+import re
+import uuid
+
+import function  # pylint:disable=import-error
+import pytest
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def test_request_no_method():
+    """Try request not containing method."""
+
+    body = {}
+
+    resp = function.lambda_handler({"body": body}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'httpMethod' is a required property."
+
+
+def test_request_no_body():
+    """Try request not containing body."""
+
+    method = "POST"
+
+    resp = function.lambda_handler({"httpMethod": method}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'body' is a required property."
+
+
+def test_post_method(db_truncate):  # pylint:disable=unused-argument
```

Comment:
A handy naming scheme for tests is to start the test name with "should" (or "test_should" if the "test_" prefix is necessary). That way the test name becomes a kind of contract, saying what the test is meant to prove rather than just the name of the thing it's testing. This is particularly useful when there are several tests for a single piece of code - rather than `test_frobnicator`, `test_frobnicator_without_foo` or something like that we end up with perhaps `should_create_instance` and `should_return_error_when_not_passing_foo`.

Reply:
Thanks, I saw your blog post on this topic :) and was expecting that you are going to raise this comment.

Reply:
@l0b0 , test names `_should_` be fixed in 212ef69d0df7bd028c5c69e490d5f23997d69cd5 :) .